### PR TITLE
SP int: ARM Thumb asm snippets

### PR DESCRIPTION
**Pull request by SparkiDev**

### CODE REVIEW

File: wolfcrypt/src/sp_int.c [hunk 1]
```diff
@@ -1115,6 +1116,1515 @@ static WC_INLINE sp_int_digit sp_div_word(sp_int_digit hi, sp_int_digit lo,
...
+#endif
+#if defined(__clang__) || defined(WOLFSSL_KEIL)
+        "subs   r3, r3, r6\n\t"
+#else
+        "sub    r3, r3, r6\n\t"
+#endif
+        "movs   %[hi], r3\n\t"
+        : [hi] "+l" (hi), [lo] "+l" (lo), [d] "+l" (d)
+        :
+        : "r3", "r4", "r5", "r6", "r7", "r8", "r9"
```
Comment (dgarske) [inline wolfcrypt/src/sp_int.c:2616]: Works in release build. With a debug build.... get the r7 issue. Not sure how you want to solve that?

```
../Middlewares/Third_Party/wolfSSL_wolfSSL_wolfSSL/wolfssl/wolfcrypt/src/sp_int.c:2619:1: error: r7 cannot be used in asm here
 2619 | }
      | ^
```

**Review by dgarske — CHANGES_REQUESTED** [review]

File: wolfcrypt/src/sp_int.c (hunk 1 above)
Comment (SparkiDev) [reply wolfcrypt/src/sp_int.c:2616]: I can't find a solution for that.
There is just too few registers available, the code would be slowed too much to support not using r7.

Comment (dgarske) [reply wolfcrypt/src/sp_int.c:2616]: Understood... perhaps you can add a note in the code around that line indicating use of release build if seeing an error on register 7?

Comment (SparkiDev) [reply wolfcrypt/src/sp_int.c:2621]: -fomit-frame-pointer dies the trick and comments added.

**Review by dgarske — APPROVED** [review]

**Pull request merged**